fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LuceneIndexScrubbingTest.java

```diff
@@ -0,0 +1,302 @@
+                    }
+                }
+            }
+            assertTrue(atLeastOnce);
+        }
+    }
+
+    @Test
+    void luceneIndexScrubMissingSimpleNoIssues() {
+        // Scrub a valid index, expect zero issues
+        Index index = SIMPLE_TEXT_SUFFIXES_WITH_PRIMARY_KEY_SEGMENT_INDEX;
+        try (final FDBRecordContext context = openContext()) {
+            // Write some records
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            recordStore.saveRecord(createSimpleDocument(1623L, ENGINEER_JOKE, 2));
+            recordStore.saveRecord(createSimpleDocument(1547L, WAYLON, 1));
+            recordStore.saveRecord(createSimpleDocument(2222L, WAYLON + " who?", 1));
+            context.commit();
+        }
+        try (final FDBRecordContext context = openContext()) {
+            // Overwrite + add records
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            recordStore.saveRecord(createSimpleDocument(1623L, ENGINEER_JOKE, 2));
+            recordStore.saveRecord(createSimpleDocument(7771547L, WAYLON, 1));
+            recordStore.saveRecord(createSimpleDocument(7772222L, WAYLON + " who?", 1));
+            context.commit();
+        }
+        try (final FDBRecordContext context = openContext()) {
+            // Scrub issues, assert none
+            rebuildIndexMetaData(context, SIMPLE_DOC, index);
+            try (OnlineIndexScrubber indexScrubber = OnlineIndexScrubber.newBuilder()
+                    .setRecordStore(recordStore)
+                    .setIndex(index)
+                    .build()) {
+                final long missingEntriesCount = indexScrubber.scrubMissingIndexEntries();
+                assertEquals(0, missingEntriesCount);
+            }
+        }
+    }
+
+
+    @ParameterizedTest
+    @MethodSource("threeBooleanArgs")
+    void luceneIndexScrubMissingDataModel(boolean isSynthetic, boolean isGrouped, boolean isPartitioned) {
+        // Scrub an index with missing entries
+        final long seed = 207L;
+
+        final LuceneIndexTestDataModel dataModel = new LuceneIndexTestDataModel.Builder(seed, this::getStoreBuilderWithRegistry, pathManager)
+                .setIsGrouped(isGrouped)
+                .setIsSynthetic(isSynthetic)
+                .setPartitionHighWatermark(isPartitioned ? 5 : 0)
+                .build();
+
+        final InjectedFailureRepository injectedFailures = new InjectedFailureRepository();
+        registry.overrideFactory(new MockedLuceneIndexMaintainerFactory(injectedFailures));
+
+        for (int i = 0; i < 14; i++) {
+            dataModel.setReverseSaveOrder(yinYang());
+            try (final FDBRecordContext context = openContext()) {
+                dataModel.saveRecords(7, context, i / 6);
+                context.commit();
+            }
+        }
+
+        try (final FDBRecordContext context = openContext()) {
+            dataModel.explicitMergeIndex(context, timer);
+            context.commit();
```

Comment:
This commit should be unnecessary, right? `explicitMergeIndex` calls the `onlineIndexer` which manages its own commits.

Reply:
Yes. This outer context should be a RO one. But the commit is harmless.